op-challenger/game/fault/types/types.go

```diff
@@ -18,14 +20,94 @@ var (
 	ErrL2BlockNumberValid = errors.New("l2 block number is valid")
 )
 
+type GameType uint32
+
+const (
```

Comment:
What's the actual difference between a game type and a trace type? Obviously TraceType has a string version we use in the CLI and game type is a uint32 from the contracts, but I _think_ they're actually the same concept, particularly since we're mapping 1:1 from trace type to game type already.

Reply:
Yeah I've been wondering as well. The only argument for keeping it is to have an immutable (const) set of trace types as opposed to `CannonGameType.String()`, etc, which would introduce type vars. Having immutables seem nicer.